packages/addon-info/src/index.tsx

```diff
@@ -1,15 +1,16 @@
 import * as React from "react";
 import { useChannel, useState } from "@storybook/client-api";
 import { TestComponent } from "./components/TestComponent";
+import {REQUEST_PROPS_CHANGE} from "./constants";
 
 export const withTestComponent = (storyFn: () => any) => {
   const [selected, setSelected] = useState<boolean>(true);
 
   const newSelect = !selected;
 
   useChannel({
-    REQUEST_PROPS_CHANGE: () => setSelected(newSelect)
-  });
+    [REQUEST_PROPS_CHANGE]: () => setSelected(newSelect)
+  }, [newSelect]);
```

Comment:
you need to pass it because otherwise the listener won't be re-attached with up-to-date `newSelect` value from closure, see https://github.com/storybookjs/storybook/pull/6916#discussion_r299400513 and yellow note from https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects